src/pptxgen.ts

```diff
@@ -560,15 +539,10 @@ export default class PptxGenJS implements IPresentationLib {
 	 * @returns {Promise<string | ArrayBuffer | Blob | Buffer | Uint8Array>} file content in selected type
 	 */
 	write(outputType: JSZIP_OUTPUT_TYPE): Promise<string | ArrayBuffer | Blob | Buffer | Uint8Array> {
-		return new Promise((resolve, reject) => {
-			this.exportPresentation(outputType)
-				.then(content => {
-					resolve(content)
-				})
-				.catch(ex => {
-					reject(ex + '\nDid you mean to use writeFile() instead?')
-				})
-		})
+		return this.exportPresentation(outputType)
+			.catch(ex => {
+				throw new Error(ex.message + '\nDid you mean to use writeFile() instead?')
```

Comment:
FWIW: This kills the stack trace of the actual exception, which seriously sucks for debugging. I would suggest to actually drop this as well, and just do:

```js
write(outputType: JSZIP_OUTPUT_TYPE): Promise<string | ArrayBuffer | Blob | Buffer | Uint8Array> {
	return this.exportPresentation(outputType)
}
```